airflow/decorators/task_group.py

```diff
@@ -132,11 +132,14 @@ def expand(self, **kwargs: OperatorExpandArgument) -> DAGNode:
         self._validate_arg_names("expand", kwargs)
         prevent_duplicates(self.partial_kwargs, kwargs, fail_reason="mapping already partial")
         expand_input = DictOfListsExpandInput(kwargs)
-        return self._create_task_group(
+        new_task_group = self._create_task_group(
             functools.partial(MappedTaskGroup, expand_input=expand_input),
             **self.partial_kwargs,
             **{k: MappedArgument(input=expand_input, key=k) for k in kwargs},
         )
+        for op, _ in expand_input.iter_references():
+            new_task_group.set_upstream(op)
```

Comment:
This logic should be in `_create_task_group` instead. The function is also called in `__call__` and this change is breaking the non-expanding use cases.